torchaudio/functional/functional.py

```diff
@@ -248,7 +251,13 @@ def amplitude_to_DB(
     x_db -= multiplier * db_multiplier
 
     if top_db is not None:
-        x_db = x_db.clamp(min=x_db.max().item() - top_db)
+        dims = x_db.dim()
+        if dims > 3:
+            db_floors = x_db.reshape(x_db.size(0), -1).amax(dim=1) - top_db
+            broadcast_shape = (...,) + (None,)*(dims-1)
+            x_db = torch.max(x_db, db_floors[broadcast_shape])
+        else:
+            x_db = x_db.clamp(min=x_db.max().item() - top_db)
```

Comment:
My suggestion is to replace directly clamp by `min and max` per batch which would simply this code.